synapse/handlers/room.py

```diff
@@ -1121,17 +1122,29 @@ async def create_event(
             )
 
         creation_content.update({"creator": creator_id})
-        creation_event, creation_context = await create_event(
-            EventTypes.Create, creation_content, False
-        )
 
         logger.debug("Sending %s in new room", EventTypes.Member)
-        ev = await self.event_creation_handler.handle_new_client_event(
-            requester=creator,
-            events_and_context=[(creation_event, creation_context)],
-            ratelimit=False,
-            ignore_shadow_ban=True,
-        )
+
+        # Try 2 times, the first one could fail with PartialStateConflictError
+        # in handle_new_client_event, cf comment in except block.
+        for _ in range(2):
+            try:
```

Comment:
Since we're in `_send_events_for_new_room`, the room should have been created by us and local-only. We can get away without the retry logic in this file.